# Goal: Use a decision tree to predict if a weapon will be involved in a crime

This will build upon previous exploration of the feasibility of a handful of models. We will go into much more detail 

In [1]:
import pandas as pd
import numpy as np
import pyspark
import numpy
from pyspark.sql import functions as F
from sklearn.model_selection import train_test_split
from sklearn.metrics import confusion_matrix, classification_report
from sklearn.tree import DecisionTreeClassifier


%matplotlib inline

In [13]:
data = pd.read_csv('gun_crime_data.csv', index_col = 'Unnamed: 0')
print('Crimes: ',len(data))
print('Kinds of crimes: ',data.description.nunique())

Crimes:  7382
Kinds of crimes:  8


Ok, we have 7382 cases of 8 different types of crimes. At least some of them involed a weapon. This should be enough to build a good predictive model.